src/hotspot/share/opto/vectornode.cpp

```diff
@@ -428,17 +432,17 @@ VectorNode* VectorNode::make_mask_node(int vopc, Node* n1, Node* n2, uint vlen,
   const TypeVect* vmask_type = TypeVect::makemask(bt, vlen);
   switch (vopc) {
     case Op_AndV:
-      if (Matcher::match_rule_supported_vector(Op_AndVMask, vlen, bt)) {
+      if (Matcher::match_rule_supported_vector_masked(Op_AndVMask, vlen, bt)) {
```

Comment:
I intended to use `match_rule_supported_vector_masked` specially for the vector operations that are controlled by a mask. Do you mean we'd better also extend it to the mask operations?

Reply:
VectorNode::make_mask_node is only going to be called for masked operations. I don't think AndVMask is being created for non-predicated targets.  So we can take either approach.

Reply:
Ok, can the opcodes like `AndVMask` be blocked for non-predicated targets in `Matcher::match_rule_supported` ?

Reply:
Matcher::match_rule_supported only check the existence of pattern based on the opcode and default return value is true. 
AddVI v1 v2
AddVI v1 v2 mask

Where as match_rule_supported_vector_masked initially calls match_rule_supported, it then return false in default case and true only for operations for which masked patterns exist.

Reply:
But I think `AndVMask/OrVMask/XorVMask` are different from the masked vector opcodes like `AddVI`. I was intending to make nodes like `AndVMask/OrVMask/XorVMask` also work for the non-mask supported platforms. Other platforms can also support them by adding the backend implementations if they want. We can just treat them as a normal vector operations for vector mask.